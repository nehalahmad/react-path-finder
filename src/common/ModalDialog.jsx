// common/ModalDialog.jsx
import React from "react";
import PropTypes from "prop-types";
import { Modal } from "react-bootstrap";
import { ERROR_TITLE } from "../config/AppConstants";

const ModalDialog = ({ errorMessage, setErrorMessageHandler }) => (
  <Modal
    show={!!errorMessage}
    centered
    onHide={() => setErrorMessageHandler("")}
  >
    <Modal.Header closeButton>
      <Modal.Title>{ERROR_TITLE}</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p>{errorMessage}</p>
    </Modal.Body>
  </Modal>
);

ModalDialog.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  setErrorMessageHandler: PropTypes.func.isRequired
};

export default ModalDialog;
