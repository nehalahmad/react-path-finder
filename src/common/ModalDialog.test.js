import React from "react";
import ModalDialog from "./ModalDialog";
import { shallow } from "enzyme";

describe("<ModalDialog />", () => {
  let setErrorMessageHandler;
  let errorMessage = "";
  beforeEach(() => {
    setErrorMessageHandler = jest.fn();
    errorMessage = "Oops! something went wrong!";
  });

  it("Render modal with a message", () => {
    const modalObj = shallow(
      <ModalDialog
        errorMessage="Oops! something went wrong!"
        setErrorMessageHandler={setErrorMessageHandler}
      />
    );
    expect(modalObj).toMatchSnapshot();
  });

  it("Render modal without a message", () => {
    errorMessage = "";
    const modalObj = shallow(
      <ModalDialog setErrorMessageHandler={setErrorMessageHandler} />
    );
    expect(modalObj).toMatchSnapshot();
  });

  it("it'll close the modal message", () => {
    const modalObj = shallow(
      <ModalDialog
        errorMessage={errorMessage}
        setErrorMessageHandler={setErrorMessageHandler}
      />
    );
    expect(modalObj).toMatchSnapshot();
  });
});
