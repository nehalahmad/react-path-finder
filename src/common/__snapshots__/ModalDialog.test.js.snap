// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDialog /> Render modal with a message 1`] = `ShallowWrapper {}`;

exports[`<ModalDialog /> it'll close the modal message 1`] = `ShallowWrapper {}`;
